### ISSUE TITLE
Easy Cast

### ISSUE THREAD

**Issue by Pantera4613** [rowengb/GSPro-MLM2PRO-OCR#2]
Hello thanks first of all for your work! 

Im having an issue with easy cast running. When I boot it up it works but after I minimize it goes to sleep and dosnt update fast enough to keep up. Any fix? Also is the MLM2 connector supposed to always be open? It shuts down after 5 min. Please help and thank you in advance.

**Comment by srbachman** [comment]
Exact same issue, I can get the first shot to work, but then EasyCast doesn't update unless I open the cast window, wait for the data to populate (5 sec), then minimize the window, then the next shot data will take, but the connector app does also auto close itself after about 5 min, and this method stops working. Thanks!

**Comment by srbachman** [comment]
> Exact same issue, I can get the first shot to work, but then EasyCast doesn't update unless I open the cast window, wait for the data to populate (5 sec), then minimize the window, then the next shot data will take, but the connector app does also auto close itself after about 5 min, and this method stops working. Thanks!

Solved the issue. Don't use EasyCast. Use scrcpy:  https://github.com/Genymobile/scrcpy
This is WAY better, no latency between phone and computer, and best of all since it uses USB to mirror, you can direct connect to the MLM2PRO which is a lot more stable. 

Once scrcpy is running check the name of the window it's running in and change settings.json to match in WINDOW_NAME. All other setup steps are the same and it works like a champ.

Only other issue is the connector sometimes repeats shots multiple times seemingly at random, ignoring the same number logic. If this happens off your Drive you can find yourself 4 shots and 500 yds past the hole.

**Comment by Pantera4613** [comment]
I cant seem to get the window name part to work. I change the code but I wrote the name and it crashed the connector. I feel like I am so close but so far at the same time. Thanks for your help!

**Comment by wonder99** [comment]
> Only other issue is the connector sometimes repeats shots multiple times seemingly at random, ignoring the same number logic. If this happens off your Drive you can find yourself 4 shots and 500 yds past the hole.

Scrcpy looks great. I'll get it going and try to debug the multiple shot issue

**Comment by wonder99** [comment]
I'm not sure wht the issue you have with scrcpy is--I had no problems. One thing though--the screen name for scrcpy was actually the name of my tablet "Lenovo xxxx".  Hover over the application in the taskbar to determine its window name--it will be very specific to you.
I've not encountered the multiple shot issue

**Comment by srbachman** [comment]
> I'm not sure wht the issue you have with scrcpy is--I had no problems. One thing though--the screen name for scrcpy was actually the name of my tablet "Lenovo xxxx". Hover over the application in the taskbar to determine its window name--it will be very specific to you. I've not encountered the multiple shot issue

Thanks for looking into it, looks like it's an issue with the OCR, see image. It appears that it is reading the vertical "MPH" on the phone setup as a character, but strange that it goes back and forth. Also it happens randomly, maybe 1 out of 20 shots. I can't set the ROI inside of the "MPH" because it does in fact move based on the value. I'll keep messing with it and my ROI's, also to see if it is always ball speed or what.

![Capture3](https://github.com/rowengb/GSPro-MLM2PRO-OCR-Connector/assets/65688338/945e8b60-205b-4f47-b96c-d1a98f305729)

**Comment by wonder99** [comment]
Yes, stay away from the vertical letters. The degree symbol is fine though. 
In an upcoming release, the box boundaries are saved between sessions so you don't need to keep doing that
